### PR TITLE
setup backend endpoints

### PR DESCRIPTION
**Pull request by mdblakestj**

### CODE REVIEW

File: backend/g_golf/admin.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (jmasonherr) [inline backend/g_golf/admin.py:3]: Nit: remove this comment

ALso, I cant comment above but it looks like you checked in the SQLIte databse. Did you intend to?

File: backend/g_golf/apps.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class GGolfConfig(AppConfig):
```
Comment (jmasonherr) [inline backend/g_golf/apps.py:4]: Typo? GGolf

File: backend/g_golf/models.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+class Hole(models.Model):
+    creator = models.ForeignKey(User, on_delete=models.CASCADE)
+    description = models.TextField()
+    name = models.CharField(max_length=100)
+    created_at = models.DateField(auto_now_add = True)
+    updated_at = models.DateField(auto_now = True)
+    par = models.PositiveIntegerField()
+    low_score = models.PositiveIntegerField()
+    long = models.DecimalField(max_digits=9, decimal_places=6)
```
Comment (jmasonherr) [inline backend/g_golf/models.py:12]: long is a reserved word in python.  this should be changed.  

Did you consider using a PointField? https://docs.djangoproject.com/en/2.2/ref/contrib/gis/model-api/#pointfield

File: backend/g_golf/models.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class Course(models.Model):
+    name = models.CharField(max_length=100)
+    creator = models.ForeignKey(User, on_delete=models.CASCADE)
+    description = models.TextField()
+    par = models.PositiveIntegerField()
+    low_score = models.PositiveIntegerField()
+    created_at = models.DateField(auto_now_add = True)
+    updated_at = models.DateField(auto_now = True)
```
Comment (jmasonherr) [inline backend/g_golf/models.py:27]: If all of these are going to have created_at, updated_at and creator, make a base model with those and then inherit from it in all of these models

File: backend/g_golf/models.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+
+    def __str__(self):
+        return self.name
+
+
+class CourseHole(models.Model):
```
Comment (jmasonherr) [inline backend/g_golf/models.py:34]: Why not just use a ManyToManyField?

File: backend/g_golf/models.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    course = models.ForeignKey(Course, on_delete=models.DO_NOTHING)
+    hole = models.ForeignKey(Hole, on_delete=models.DO_NOTHING)
+
+
+
+
+class Game(models.Model):
+    creator = models.ForeignKey(User, on_delete=models.CASCADE)
+    course = models.ForeignKey(Course, on_delete=models.CASCADE)
+    player_2_name = models.CharField(max_length=100)
```
Comment (jmasonherr) [inline backend/g_golf/models.py:44]: Missing player 1 name? Its more inline with relational database modeling standards to store a a GamePlayer and setting up a relationship to Game instead of the player_x_... design, but its NBD. Just want to suggest the option if you're unhappy with this setup. I suspect this will lead to a lot of checking of how many players, etc... and it may be easier to deal with the users as a list.  If you do stick with this design the 'correct' way would be storing the user primary key as a foreign key field

File: backend/g_golf/models.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    created_at = models.DateField(auto_now_add = True)
+    updated_at = models.DateField(auto_now = True)
+    present_hole = models.PositiveIntegerField(default=1)
+
+
+    def __str__(self):
+        return self.created_at
+
+
+class HoleInst(models.Model):
```
Comment (jmasonherr) [inline backend/g_golf/models.py:60]: Nit: Docstring.  What is a HoleInst? Is it a person's score at that hole? All people's score at that hole? Again I think you'll be better off modeling the user relationship to that hole, score and game instead of making a list like this.

File: backend/g_golf/serializers.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+
+
+
+class CourseHoleSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CourseHole
+        fields = '__all__'
+
+    # def create(self, validated_data):
```
Comment (jmasonherr) [inline backend/g_golf/serializers.py:49]: Nit: Why is this here? If a todo, mark with # TODO(yourusername): fix this do whatever etc.

File: backend/g_golf/views.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+class CourseHoleView(viewsets.ModelViewSet):
+    queryset = CourseHole.objects.all()
+    serializer_class = CourseHoleSerializer
+
+class CourseView(viewsets.ModelViewSet):
+    queryset = Course.objects.all()
+    serializer_class = CourseSerializer
+
+# Create your views here.
```
Comment (jmasonherr) [inline backend/g_golf/views.py:35]: Nit: delete comment

File: backend/golf/settings.py [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+import os
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
```
Comment (jmasonherr) [inline backend/golf/settings.py:22]: lololol.  find a way to keep this secret and don't check it into Git.

File: backend/golf/settings.py [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+SECRET_KEY = 'cu3-m6a)1wz2!6a*ik0ny8nr)$+2e(rz8y^qjb6uv!3kn4tzts'
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (jmasonherr) [inline backend/golf/settings.py:26]: You'll need a way to gate this depening on the environment you're running

File: backend/golf/settings.py [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+Generated by 'django-admin startproject' using Django 2.2.6.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/2.2/ref/settings/
+"""
+
```
Comment (jmasonherr) [inline backend/golf/settings.py:12]: You're goint to want some kind of variables to track the environment this is running in

File: backend/golf/settings.py [hunk 13]
```diff
@@ -0,0 +1,124 @@
+"""
```
Comment (jmasonherr) [inline backend/golf/settings.py:1]: This should be part of a separate PR.  Try to group them by functionality, and break them into as small of pieces as possible.  Lots of tiny reviews > big reviews

File: backend/golf/urls.py [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+Function views
+    1. Add an import:  from my_app import views
+    2. Add a URL to urlpatterns:  path('', views.home, name='home')
+Class-based views
+    1. Add an import:  from other_app.views import Home
+    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
+Including another URLconf
+    1. Import the include() function: from django.urls import include, path
+    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
+"""
```
Comment (jmasonherr) [inline backend/golf/urls.py:15]: Delete these comments, replace with your own if you hav esomething helpful to say

File: backend/golf/settings.py [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+
+    'g_golf.apps.GGolfConfig',
```
Comment (jmasonherr) [inline backend/golf/settings.py:41]: Its nice to add sections in this as it grows

`# 3rd party libraries`
...
`# REST dependencies`
...

**Review by jmasonherr — COMMENTED** [review]
Cool, you should probably break this into smaller reviews  that address configuration and setup and address models.  You may want to denormalize your database models

File: backend/g_golf/models.py (hunk 6 above)
Comment (mdblakestj) [reply backend/g_golf/models.py:44]: The issue here is that only one player will be in the database. The creator of the game signs up with the app but all other players just have a short name with no other information.

File: backend/g_golf/models.py (hunk 7 above)
Comment (mdblakestj) [reply backend/g_golf/models.py:60]: A hole inst is the score for a particular hole as the players are playing it. The issue is that all other players besides the person who signed in don't have a user model. I want players 2-4 just to be able to put in a short name at least for the mvp.
